ExtractionStrategy/DefaultExtractionStrategies.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Innmind\Reflection\ExtractionStrategy;
+
+use Innmind\Immutable\TypedCollection;
+use Innmind\Reflection\Cache\StrategyCachingCapabilities;
+use Innmind\Reflection\Exception\LogicException;
+
+/**
+ * DefaultExtractionStrategies
+ *
+ * @author Hugues Maignol <hugues@hmlb.fr>
+ */
+final class DefaultExtractionStrategies implements ExtractionStrategies
+{
+
+    use StrategyCachingCapabilities;
+
+    private $strategies;
+
+    public function all() : TypedCollection
```

Comment:
as of [psr 12](https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#example)  the return type should be written like this `method(): hint` (no space between closing parenthesis and the colon)